aten/src/ATen/TensorBase.h

```diff
@@ -1,16 +1,19 @@
 #pragma once
 
 #include "ATen/TensorImpl.h"
+#include "ATen/UndefinedTensor.h"
 
 namespace at { namespace detail {
 
 // TensorBase is the base class for Tensor which handles the reference counting
 struct TensorBase {
-  TensorBase()
-  : pImpl(nullptr) {}
+  TensorBase(): TensorBase(UndefinedTensor::singleton(), true) {}
```

Comment:
Yes, it seemed easier that way because we don't really have a mechanism to track whether something shouldn't be refcounted (besides retain in the TensorImpl constructor, but that isn't maintained across copies).